sqeleton/databases/snowflake.py

```diff
@@ -163,26 +163,44 @@ def __init__(self, *, schema: str, **kw):
 
         assert '"' not in schema, "Schema name should not contain quotes!"
         # If a private key is used, read it from the specified path and pass it as "private_key" to the connector.
-        if "key" in kw:
-            with open(kw.get("key"), "rb") as key:
-                if "password" in kw:
-                    raise ConnectError("Cannot use password and key at the same time")
-                p_key = serialization.load_pem_private_key(
-                    key.read(),
-                    password=None,
-                    backend=default_backend(),
-                )
-
-            kw["private_key"] = p_key.private_bytes(
-                encoding=serialization.Encoding.DER,
-                format=serialization.PrivateFormat.PKCS8,
-                encryption_algorithm=serialization.NoEncryption(),
-            )
-
+        if "key" in kw or "key_path" in kw:
```

Comment:
Hey @jakedave I'm just realizing that dbt only supports a private_key_path, there's no need to add any support for a string

Are you able to rework this with the previous assumption? `kw.get("key")` <--- path

We shouldn't stray from what is defined here https://docs.getdbt.com/reference/warehouse-setups/snowflake-setup#key-pair-authentication

Reply:
Once that's done we can release a new sqeleton version, add the new sqeleton version in the data-diff PR, and remove this method there
```
    @staticmethod
    def _get_snowflake_private_key(credentials):
```